src/back/hlsl/writer.rs

```diff
@@ -1215,11 +1237,175 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                     writeln!(self.out, "[_i] = _result[_i];")?;
                     writeln!(self.out, "{}}}", level)?;
                 } else {
+                    // We treat matrices of the form matCx2 as a sequence of C vec2's due to
```

Comment:
please make sure to write something down at the top doc comment of `mod.rs` about this